src/client/connection.ts

```diff
@@ -244,8 +241,9 @@ export class Connection extends EventEmitter {
     this.ws.on('error', () => clearTimeout(connectionTimeoutID))
     this.ws.on('close', (reason) => this.onConnectionFailed(reason))
     this.ws.on('close', () => clearTimeout(connectionTimeoutID))
-    // eslint-disable-next-line @typescript-eslint/no-misused-promises -- TODO: resolve this
-    this.ws.once('open', async () => this.onceOpen(connectionTimeoutID))
+    this.ws.once('open', () => {
```

Comment:
Spawn these promises as `void`

Reply:
There's nothing awaiting them, so this should be fine.

Reply:
what's the purpose of using `void` here?

Reply:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md#ignorevoid